### PR TITLE
Refactor FXIOS-5402 [v113] Remove Show Tour setting option

### PR DESCRIPTION
**Pull request by nishant2718**
# [FXIOS-5402](https://mozilla-hub.atlassian.net/browse/FXIOS-5402)

TLDR: ~Remove the foregroundBVC pattern from the app  🥳~ Remove show tour settings option.

### CODE REVIEW

File: Client/Frontend/TabContentsScripts/DownloadContentScript.swift [hunk 1]
```diff
@@ -56,12 +56,7 @@ class DownloadContentScript: TabContentScript {
               let data = Bytes.decodeBase64(base64String)
         else { return }
 
-        // TODO: Could we have a download queue per tab instead of resolving from foregroundBVC?
-        // Or one that is independent of BVC at least?
-        let browserViewController = BrowserViewController.foregroundBVC()
-
         defer {
-            browserViewController?.pendingDownloadWebView = nil
```
Comment (OrlaM) [inline Client/Frontend/TabContentsScripts/DownloadContentScript.swift:64]: How come this isn't needed?

Comment (nishant2718) [reply Client/Frontend/TabContentsScripts/DownloadContentScript.swift:64]: I believe `pendingDownloadWebView` gets reset every time in [BVC+WebViewDelegates](https://github.com/mozilla-mobile/firefox-ios/blob/4d4876781e880a582deab565db901d2ce1cf19eb/Client/Frontend/Browser/BrowserViewController/BrowserViewController%2BWebViewDelegates.swift#L673).

**Comment by mte-test** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>Project coverage: 30.13%</td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>Edited 7 files</td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>Created 0 files</td>
    </tr>
  </tbody>
</table>

## Client.app: Coverage: 27.01
| File | Coverage ||
| --- | --- | --- |
AppSettingsTableViewController.swift | 0.0% | ⚠️
DownloadContentScript.swift | 0.0% | ⚠️
AppSettingsOptions.swift | 0.0% | ⚠️
BrowserViewController.swift | 20.72% | ⚠️
TelemetryWrapper.swift | 66.7% | ✅

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/swift">Danger Swift</a> against 520b45f68d27770cee8b35909b1e87c70ea838c4
</p>

File: Client/Frontend/TabContentsScripts/DownloadContentScript.swift (hunk 1 above)
Comment (OrlaM) [reply Client/Frontend/TabContentsScripts/DownloadContentScript.swift:64]: Do you know how to trigger this code path? I couldn't hit it by just downloading a random file.

Comment (nishant2718) [reply Client/Frontend/TabContentsScripts/DownloadContentScript.swift:64]: I tried a lot but can't really find a way to trigger it. But I made sure for each download, the `pendingDownloadWebView` does get `nil`ed out prior to the next one starting.

Comment (OrlaM) [reply Client/Frontend/TabContentsScripts/DownloadContentScript.swift:64]: @nbhasin2 Do you know how to trigger this code path? I tried a bunch of different download types from both clicking links and the long press menu and couldn't trigger it. 

I'm not super comfortable with this change if we don't understand what it's doing. If it is just a dead code path tho then we should create a follow up task to remove it altogether.

Comment (nbhasin2) [reply Client/Frontend/TabContentsScripts/DownloadContentScript.swift:64]: Let me try a flow that might work and revert back here

Comment (nbhasin2) [reply Client/Frontend/TabContentsScripts/DownloadContentScript.swift:64]: This is for `blob` downloads and should not be removed unless you are replacing this with something equally in line with the change.

image

image

**Review by nbhasin2 — CHANGES_REQUESTED** [review]
Do not remove blob downloads unless there is a justification on why we are removing it

File: Client/Frontend/TabContentsScripts/DownloadContentScript.swift (hunk 1 above)
Comment (nbhasin2) [reply Client/Frontend/TabContentsScripts/DownloadContentScript.swift:64]: You can try it by long pressing on a valid URL `blob:https://....`
Simple google should be able to get you some download URL websites to test blob download

Comment (nishant2718) [reply Client/Frontend/TabContentsScripts/DownloadContentScript.swift:64]: Gotcha - I'll undo that particular change.

**Review by OrlaM — APPROVED** [review]

**Review by nbhasin2 — APPROVED** [review]

**Pull request merged**